feat: implemented smart redirect

captainPrime opened the pull request:
# Description

This pull request implements a feature that redirects the user back to the page they were on after they sign in. The feature uses Vue.js, Vuex, and vue-router to store the redirect path before the login action and then redirects the user to the stored path after successful login.

# How Has This Been Tested?

I have tested this feature using the following steps:

1. Set up a Vue.js project with Vue Router and Vuex.
2. used the signin component that stores the current route path before login and redirects the user back to the stored path after successful login.
3. Manually tested the sigin and redirection functionality by navigating to different routes and logging in.
4. Created an end-to-end (E2E) test using Nightwatch.js to verify the login and redirection functionality automatically.
5. Ensured that the E2E test passes successfully by running `npx nightwatch`.

- [✅] Manual tested on my local
- [✅ ] e2e test

# Checklist:
- [ ] I have rebased my branch against the base branch
- [✅] My code follows the style guidelines of this project (4 spaces instead of 2)
- [✅] I have performed a self-review of my code. 
- [✅] I have cleaned my commits

fokosun commented on src/store/index.js:
```diff
@@ -126,7 +128,10 @@ export default new Vuex.Store({
 			this.state.active_user.onboarding.likehihoodToShare = choice
-		}
+		},
+		setRedirectPath(state, path) {
```
This is an action right, it does not belong here under mutations. Leave move it to actions object. Also use snake_case for consistency

fokosun commented on src/components/Navigation.vue:
```diff
@@ -60,7 +59,11 @@ export default {
+		},
+		navigateToSignin() {
+			this.$store.commit('setRedirectPath', this.$route.path);
+			VueRouter.push('/signin');
```
I believe you meant to handle this with the dispatched event. There is no need to import vuerouter into this component, the router is already available for use everywhere. Remove this line and the import statement above. Inside your setRedirectPath action, use a mutation to redirect user to the signin page

fokosun commented on src/components/Register.vue:
```diff
@@ -54,11 +54,18 @@ export default {
-                window.location.href = process.env.APP_URL + '/#/' + provider + '/?code=&_d='
+            //window.location.href = process.env.APP_URL + '/#/' + provider + '/?code=&_d='
+
+            setTimeout(() => {
```
Does this belong here? You are completely overriding the login behavior in dev.

fokosun commented on src/components/Register.vue:
```diff
@@ -81,8 +88,10 @@ export default {
                 }
-
-                window.location.href = url;
+                setTimeout(() => {
```
What does this do? Why not abstract this to a function or an action instead of duplicating code?

fokosun replied:
https://router.vuejs.org/guide/advanced/navigation-guards.html#optional-third-argument-next

fokosun commented:
I didn't see the logic that remembers where the user was and redirects them back after login. Did I miss it? Also, I didnt see the logic that handles redirecting the user to the page they were if they happen to logout due to expired session and log back in. Depending on where they were, the request should also contain unsubmitted form data if they were on a page that has a form.

fokosun commented:
Converted back to draft since it is still a WIP

captainPrime commented:
I

captainPrime commented:
> I didn't see the logic that remembers where the user was and redirects them back after login. Did I miss it? Also, I didnt see the logic that handles redirecting the user to the page they were if they happen to logout due to expired session and log back in. Depending on where they were, the request should also contain unsubmitted form data if they were on a page that has a form. So fir example they were creating a recipe, then they were primpted to log out and log in again, the request should remember both the route they were and the form payload so that the frontend can use the data to repopulate the form again so they dont have to start from scratch again.
> 
> https://stackoverflow.com/questions/53788975/vue-router-how-to-get-previous-page-url https://prismic.io/docs/vue-route-parameters
> 
> I was expecting a middleware or some implementation not tightly coupled to any component, to make it reuseable everywhere and just work automatically right out of the box.

I refactored the approach to this feature... updated the auth middleware to handle saving the previous page and redirecting after login
